### PR TITLE
add Updates missing APIs

### PR DESCRIPTION
**Pull request by gtalha07**
Refs: #793 -> Add Updates missing APIs and structs for front-end

- [x] Add ```NewsSlideDraft```, which is automatically created and pushed in draft list by ```fn add_slide()``` in ```NewsEntryDraft```.
- [x] Add ```fn insert_slide(pos: int)``` which can be used for adding and arranging slide drafts.
- [x] Expose ```fn colors()``` for ```NewsEntry```.
- [x] Add and expose ```fn slides()``` for fetching all slides of ```NewsEntry```. 
- [x] Updated flutter code to support horizontal scrolling on ```NewsEntry``` having more slides.
- [x] Added rust integration test for multiple slide draft send and fetch.
- [x] Fix ```NewsEntryUpdateBuilder``` for editing updates.

### CODE REVIEW

**Review by kumarpalsinh25 — APPROVED** [review]
Looks good flutter side changes!

**Comment by gnunicorn** [comment]
@gtalha07 CI is unhappy...

File: native/acter/src/api/news.rs [hunk 1]
```diff
@@ -292,6 +292,20 @@ impl NewsSlide {
     }
 }
 
+#[derive(Clone)]
+pub struct NewsSlideDraft {
+    client: Client,
+    room: Room,
```
[the remark this answers is not in the document]
Comment (gtalha07) [reply native/acter/src/api/news.rs:298]: Ahh forgot to remove that, it was added during improvising....let me update it.

File: native/acter/src/api/news.rs [hunk 2]
```diff
@@ -400,33 +428,54 @@ impl NewsEntry {
...
-        self.slides.push(slide);
+        self.slides.push(inner);
+        Ok(true)
+    }
+
+    pub async fn insert_slide(
+        &mut self,
+        pos: u8,
+        base_draft: Box<MsgContentDraft>,
+    ) -> Result<bool> {
```
[the remark this answers is not in the document]
Comment (gtalha07) [reply native/acter/src/api/news.rs:464]: The thought here was maybe we want to re-arrange slide or adding the slide in already growing list by ordering it differently?

[the remark this answers is not in the document]
Comment (gnunicorn) [reply native/acter/src/api/news.rs:462]: but for reordering you have `swap` no? The insert doesn't really make sense to me...

[the remark this answers is not in the document]
Comment (gtalha07) [reply native/acter/src/api/news.rs:462]: Hmmm, yes you're correct, it seems more sensible to change this fn to handle ordering of  ```NewsEntryDraft``` inner object ```Vec<NewsSlideDraft>```.

**Pull request merged**